File: archive/Backend/app/services/validation_service.py
from __future__ import annotations

import hmac

from app.config import settings


def validate_token(token: str | None) -> bool:
    if not token:
        return False
    return hmac.compare_digest(token, settings.jotform_webhook_secret)


def validate_form_id(form_id: str) -> bool:
    return form_id == settings.jotform_form_id

File: Backend-js/src/services/validation_service.ts
import { timingSafeEqual } from 'node:crypto';
import { settings } from '../config.js';

export function validateToken(token: string | undefined | null): boolean {
  if (!token) return false;
  // Constant-time compare to match the Python hmac.compare_digest semantics.
  const a = Buffer.from(token);
  const b = Buffer.from(settings.jotform_webhook_secret);
  if (a.length !== b.length) return false;
  return timingSafeEqual(a, b);
}

export function validateFormId(formId: string): boolean {
  return formId === settings.jotform_form_id;
}
